pages/movies/[id].js

```diff
@@ -0,0 +1,86 @@
+import Head from "next/head";
+import Image from "next/image";
+import styles from "../../styles/Movie.module.css";
+
+import Navbar from "../../components/header/Navbar";
+import Logo from "../../components/header/Logo";
+import Banner from "../../components/header/Banner";
+import RightSidebar from "../../components/RightSidebar";
+
+import connectDb from "../../DB/connectDb";
+import Movies from "../../DB/models/movies";
+import Screenings from "../../DB/models/screenings";
+
+// Get requested movie and screenings for that movie from DB
+export async function getServerSideProps(context) {
+  await connectDb();
+  const movieID = JSON.parse(`{ "movieid":${context.query.id} }`);
+
+  const movie = await Movies.findOne(context.query, { _id: 0 }).lean();
+  const screenings = await Screenings.find(movieID, { _id: 0 }).lean();
+
+  if (!movie) {
+    return { notFound: true };
+  }
+  return { props: { movie, screenings } };
+}
+
+export default function Post({ movie, screenings }) {
```

Comment:
I think the named `Post` is used in some NEXT examples, so I appreciate where this is coming from, but I would use something more descriptive of this page, e.g. `MoviePage`.